### PR TITLE
Added basic highscore tracking for scrolly-tile

### PR DESCRIPTION
**Pull request by ThePythonator**
Daft-Freak mentioned this a while ago, that highscore tracking would be cool.
This adds basic highscore tracking to scroll-tile (ignoring the level seed).

### CODE REVIEW

File: examples/scrolly-tile/scrolly-tile.cpp [hunk 1]
```diff
@@ -107,6 +112,10 @@ using tile_callback = uint8_t (*)(uint8_t tile, uint8_t x, uint8_t y, void *args
 uint32_t prng_lfsr = 0;
 const uint16_t prng_tap = 0x74b8;
 
+uint32_t max(uint32_t a, uint32_t b) {
+  return a > b ? a : b;
+}
+
```
Comment (Daft-Freak) [inline examples/scrolly-tile/scrolly-tile.cpp:118]: Think you want `std::max` from `#include <algorithm>`

Comment (ThePythonator) [reply examples/scrolly-tile/scrolly-tile.cpp:118]: Didn't know that existed! 😄

Comment (Gadgetoid) [reply examples/scrolly-tile/scrolly-tile.cpp:118]: You can probably drop this, plus the "highscore" variable altogether and just do:

```c++
if (player_progress > saveData.highscore) {
    saveData.highscore = player_progress;
}
```

No need to keep another copy of the score, the struct has you covered.

Comment (Daft-Freak) [reply examples/scrolly-tile/scrolly-tile.cpp:118]: Hmm, yeah... should've read the rest! Anyway, that should really be `save_data` for consistency.

Comment (ThePythonator) [reply examples/scrolly-tile/scrolly-tile.cpp:118]: Need to update it when displaying the highscore as well then

Comment (ThePythonator) [reply examples/scrolly-tile/scrolly-tile.cpp:118]: I'll fix this tomorrow I think

**Comment by Gadgetoid** [comment]
Awesome, thank you! Will aim to get this tested & merged ASAP since I want to tag a new release SOON. Somewhat caught up in Sprite Editor :laughing:

**Comment by Gadgetoid** [comment]
Why did I make this awful hard game? :laughing: 

Thank you! Works a treat.

**Pull request merged**